### PR TITLE
Add logsdb telemetry

### PR DESCRIPTION
**Pull request by dnhatn**
This PR adds telemetry for logsdb. However, this change only tracks the count of indices using logsdb and those that use synthetic source. Additional stats, such as shard, indexing, and search stats, will be added in a follow-up, as they require reaching out to data nodes.

### CODE REVIEW

File: x-pack/plugin/logsdb/src/main/java/org/elasticsearch/xpack/logsdb/LogsDBUsageTransportAction.java [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+            clusterService,
+            threadPool,
+            actionFilters,
+            indexNameExpressionResolver
+        );
+        this.clusterService = clusterService;
+    }
+
+    @Override
+    protected void masterOperation(
```
Comment (martijnvg) [inline x-pack/plugin/logsdb/src/main/java/org/elasticsearch/xpack/logsdb/LogsDBUsageTransportAction.java:52]: I don't know if this is the best place for this, but can we maybe in a follow up also track the number of indices without index mode that use synthetic source, the number of tsdb indices and how many of those tsdb indices use synthetic source? If you think this is a good idea. Maybe we should in this change rename the `indices_count` and `indices_with_synthetic_source` fields in the response to be something like `logsdb_index_count`, `logsdb_with_synthetic_source_index_count`?

File: x-pack/plugin/logsdb/src/main/java/org/elasticsearch/xpack/logsdb/LogsDBUsageTransportAction.java [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+        Task task,
+        XPackUsageRequest request,
+        ClusterState state,
+        ActionListener<XPackUsageFeatureResponse> listener
+    ) {
+        int numIndices = 0;
+        int numSyntheticSources = 0;
```
Comment (martijnvg) [inline x-pack/plugin/logsdb/src/main/java/org/elasticsearch/xpack/logsdb/LogsDBUsageTransportAction.java:59]: nit: maybe rename to `numLogsdbIndices` and `numLogsdbIndicesWithSynthticSource`?

**Review by martijnvg — APPROVED** [review]
LGTM. 

I think we also need to update the mapping the telemetry repo. Similar to this elastic/telemetry#3415

File: x-pack/plugin/logsdb/src/main/java/org/elasticsearch/xpack/logsdb/LogsDBUsageTransportAction.java (hunk 1 above)
Comment (dnhatn) [reply x-pack/plugin/logsdb/src/main/java/org/elasticsearch/xpack/logsdb/LogsDBUsageTransportAction.java:52]: Yes, I considered this as well, but these fields don't seem to fit within logsdb. I think we'll need to add two separate usages: one for time_series and another for source_mode. I’ll add them in follow-up PRs.

```
{
	"logsdb": {
		"enabled": true,
		"available": true,
		"indices_count": 1,
		"indices_with_synthetic_source": 1
	}
}
```

File: x-pack/plugin/logsdb/src/main/java/org/elasticsearch/xpack/logsdb/LogsDBUsageTransportAction.java (hunk 2 above)
Comment (dnhatn) [reply x-pack/plugin/logsdb/src/main/java/org/elasticsearch/xpack/logsdb/LogsDBUsageTransportAction.java]: I pushed https://github.com/elastic/elasticsearch/pull/115994/commits/5dc58e66fab6e87f08fba463de052156ec7cd707

Comment (martijnvg) [reply x-pack/plugin/logsdb/src/main/java/org/elasticsearch/xpack/logsdb/LogsDBUsageTransportAction.java]: 👍

**Comment by dnhatn** [comment]
Thanks Martijn!

**Pull request merged**